src/components/InventoryV2/form/useSubLocations.ts

```diff
@@ -37,10 +37,8 @@ export const useSubLocations = (nurseryId?: number, record?: { subLocationIds?:
   }, [availableSubLocations, record?.subLocationIds]);
 
   useEffect(() => {
-    if (!availableSubLocations) {
-      void initSubLocations();
-    }
-  }, [availableSubLocations, initSubLocations]);
+    void initSubLocations();
```

Comment:
initSublocations callback changes when nursery id or record's facility id changes, we need to refetch sublocations when either of those change